safenode/src/bin/kadnode.rs

```diff
@@ -28,12 +27,33 @@ async fn main() -> Result<()> {
     let _log_appender_guard = init_node_logging(&opt.log_dir)?;
 
     let socket_addr = SocketAddr::new(opt.ip, opt.port);
+    let peers = parse_peer_multiaddreses(&opt.peers)?;
 
     info!("Starting a node...");
-    let (_node, node_events_channel) = Node::run(socket_addr).await?;
+    let (node, node_events_channel) = Node::run(socket_addr).await?;
 
     let mut node_events_rx = node_events_channel.subscribe();
-    if let Ok(event) = node_events_rx.recv().await {
+
+    for (peer_id, addr) in &peers {
+        // TODO: Do we want to wait for the dial?
+        if let Err(err) = node.network.dial(*peer_id, addr.clone()).await {
```

Comment:
as here is the `initial` dials, I think here shall wait for the dial result.
However, as the reading in list may contain outdated peer, may better set a time out for it.
AND, the initial dial process may can be completed on the first (or first X) successful dials?